arch/Kconfig

```diff
@@ -40,6 +40,7 @@ config X86
 	select HAS_DTS
 	select ARCH_HAS_CUSTOM_SWAP_TO_MAIN if !X86_64
 	select CPU_HAS_MMU
+	select ARCH_HAS_GDB_SUPPORT
```

Comment:
GDB_SUPPORT is too broad and might be confusion, we probably should be specific here and replace it with GDB_SERVER or GDB_STUB